logparser/devices/logger.py

```diff
@@ -0,0 +1,166 @@
+    if state['verbosity'] < level:
+        return
+
+    # Add the clock if available
+    if 'clocks' in state and state['clocks'][1]:
+        content['timestamp'] = " %s " % state['clocks'][1].isoformat()
+
+    # Add the current line
+    content['input_line'] = state['input_line']
+    content['output_line'] = state['output_line'] + 1  # This message count
+
+    # Apply the filter
+    if 'onlyIf' in state and not dict_regex_search(content, state['onlyIf']):
+        return
+
+    # Highlight the message if match
+    if 'highlight' in state and dict_regex_search(content, state['highlight']):
+        content['kind'] = content.get('kind', "") + "|IMPORTANT"
+
+    # Apply color if specified
+    if not state['no_colors']:
+        color = ""
+        for kind in filter(None, content.get('kind', '').split("|")):
+            for subkind in KIND_TO_COLOR[kind].split("|"):
+                color += COLORS[subkind]
+        if len(color) > 0:
+            content['description'] = color + content['description'] + \
+                COLORS['END']
+
+    # Write the message
+    state['format_device'].write_message(content, state)
+
+
+def log_recv(addr, entity, text, state, level=0):
+    """Log a received packet."""
+    if state['ignore_packets']:
+        return
+    content = {'description': text, 'remote': addr, 'entity': entity,
+               'inout': 'in'}
+    log(content, level, state)
+
+
+def log_send(addr, entity, text, state, level=0):
+    """Log a sent packet."""
+    if state['ignore_packets']:
+        return
+    content = {'description': text, 'remote': addr, 'entity': entity,
+               'inout': 'out'}
+    log(content, level, state)
+
+
+def log_process(addr, entity, text, state, level=0):
+    """Log a processed packet."""
+    if state['ignore_packets']:
+        return
+    content = {'description': text, 'remote': addr, 'entity': entity}
+    log(content, level, state)
+
+
+def log_cfg(text, state, level=0):
+    """Log a configuration message."""
+    if state['verbosity'] < level:
+        return
+    countset_add_element(state['config'], text)
+
+
+def log_event(text, state, level=0):
+    """Log an application event."""
+    content = {'description': text}
+    log(content, level, state)
+
+
+def log_warning(text, state, level=0):
+    """Log a warning message."""
+    if state['verbosity'] < level:
+        return
+
+    countset_add_element(state['warnings'], text)
+    if state['inline']:
+        content = {'description': "Warning: " + text, 'kind': 'WARNING'}
+        log(content, level, state)
+
+
+def log_error(text, state, level=0):
+    """Log an error."""
+    if state['verbosity'] < level:
+        return
+
+    countset_add_element(state['errors'], text)
+    if state['inline']:
+        content = {'description': "Error: " + text, 'kind': 'ERROR'}
+        log(content, level, state)
+
+
+def countset_add_element(countset, el):
+    """Add an element to the countset."""
+    if el not in countset:
```

Comment:
You are doing this because you don't want have "el" repeated in the countset, true? So, we should use a set, not a list:
https://docs.python.org/2/library/stdtypes.html#set

(countset --> count-set --> count set. Oh! Wait!! It is a set!) :dancer: